moveit_planners/ompl/ompl_interface/src/parameterization/work_space/pose_model_state_space.cpp

```diff
@@ -131,9 +133,31 @@ void PoseModelStateSpace::sanityChecks() const
 void PoseModelStateSpace::interpolate(const ompl::base::State* from, const ompl::base::State* to, const double t,
                                       ompl::base::State* state) const
 {
-  // interpolate in joint space
+  // we want to interpolate in Cartesian space to avoid rejection of path constraints
+
+  // interpolate in joint space to find a suitable seed for IK
   ModelBasedStateSpace::interpolate(from, to, t, state);
-  computeStateFK(state);
+  double d_joint = ModelBasedStateSpace::distance(from, state);
+
+  // interpolate SE3 components
+  for (std::size_t i = 0; i < poses_.size(); ++i)
```

Comment:
It seems we're not using `i` anywhere but to index `poses`, so this can be a range based for-loop.

Reply:
This all just restored previous code. I didn't improve it. Please feel free doing it.